<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>719</width>
    <height>837</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>719</width>
    <height>837</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>719</width>
    <height>837</height>
   </size>
  </property>
  <property name="font">
   <font>
    <pointsize>11</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>Лабораторная работа №1</string>
  </property>
  <property name="windowOpacity">
   <double>1.000000000000000</double>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="font">
    <font>
     <pointsize>11</pointsize>
    </font>
   </property>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>50</y>
      <width>521</width>
      <height>171</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab">
     <property name="accessibleName">
      <string>Перемещение</string>
     </property>
     <attribute name="title">
      <string>Перемещение</string>
     </attribute>
     <widget class="QFrame" name="move_frame">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>261</width>
        <height>141</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>31</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>dx = </string>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>31</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>dy = </string>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>35</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>dz =</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="move_x">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>40</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="inputMask">
        <string/>
       </property>
       <property name="text">
        <string>0.0</string>
       </property>
       <property name="placeholderText">
        <string>Х смещение</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="move_y">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>70</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>0.0</string>
       </property>
       <property name="placeholderText">
        <string>Y смещение</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="move_z">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>100</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>0.0</string>
       </property>
       <property name="placeholderText">
        <string>Z смещение</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>5</y>
         <width>241</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Смещение</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Масштабирование</string>
     </attribute>
     <widget class="QFrame" name="move_frame_2">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>251</width>
        <height>141</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>31</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>x = </string>
       </property>
      </widget>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>31</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>y = </string>
       </property>
      </widget>
      <widget class="QLabel" name="label_8">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>35</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>z =</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="scale_x">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>40</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="inputMask">
        <string/>
       </property>
       <property name="text">
        <string>0.0</string>
       </property>
       <property name="placeholderText">
        <string>Х центр</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="scale_y">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>70</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>0.0</string>
       </property>
       <property name="placeholderText">
        <string>Y центр</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="scale_z">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>100</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>0.0</string>
       </property>
       <property name="placeholderText">
        <string>Z центр</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_9">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>5</y>
         <width>241</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Центр масштабирования</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </widget>
     <widget class="QFrame" name="move_frame_4">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>0</y>
        <width>251</width>
        <height>141</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QLabel" name="label_14">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>31</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>kx = </string>
       </property>
      </widget>
      <widget class="QLabel" name="label_15">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>31</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>ky = </string>
       </property>
      </widget>
      <widget class="QLabel" name="label_16">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>35</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>kz =</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="scale_kx">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>40</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="inputMask">
        <string/>
       </property>
       <property name="text">
        <string>1.0</string>
       </property>
       <property name="placeholderText">
        <string>kX</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="scale_ky">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>70</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>1.0</string>
       </property>
       <property name="placeholderText">
        <string>kY</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="scale_kz">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>100</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>1.0</string>
       </property>
       <property name="placeholderText">
        <string>kZ</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_17">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>5</y>
         <width>241</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Коэффициенты масштабирования</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </widget>
     <widget class="Line" name="line">
      <property name="geometry">
       <rect>
        <x>239</x>
        <y>-10</y>
        <width>31</width>
        <height>181</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>Вращение</string>
     </attribute>
     <widget class="QFrame" name="move_frame_3">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>251</width>
        <height>141</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>31</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>x = </string>
       </property>
      </widget>
      <widget class="QLabel" name="label_11">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>31</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>y = </string>
       </property>
      </widget>
      <widget class="QLabel" name="label_12">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>35</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>z =</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="rotate_x">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>40</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="inputMask">
        <string/>
       </property>
       <property name="text">
        <string>0.0</string>
       </property>
       <property name="placeholderText">
        <string>Х центр</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="rotate_y">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>70</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>0.0</string>
       </property>
       <property name="placeholderText">
        <string>Y центр</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="rotate_z">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>100</y>
         <width>181</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>0.0</string>
       </property>
       <property name="placeholderText">
        <string>Z центр</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_13">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>5</y>
         <width>241</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Центр вращения</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </widget>
     <widget class="Line" name="line_2">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>-10</y>
        <width>31</width>
        <height>191</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </widget>
     <widget class="QFrame" name="move_frame_5">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>0</y>
        <width>251</width>
        <height>141</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QLabel" name="label_18">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>41</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>∠x = </string>
       </property>
      </widget>
      <widget class="QLabel" name="label_20">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>41</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>∠z =</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="rotate_ax">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>40</y>
         <width>161</width>
         <height>20</height>
        </rect>
       </property>
       <property name="inputMask">
        <string/>
       </property>
       <property name="text">
        <string>0.0</string>
       </property>
       <property name="placeholderText">
        <string>Вращение вокруг OX</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="rotate_az">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>100</y>
         <width>161</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>0.0</string>
       </property>
       <property name="placeholderText">
        <string>Вращение вокруг OZ</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_21">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>5</y>
         <width>241</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Углы вращения</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_19">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>41</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>∠y = </string>
       </property>
      </widget>
      <widget class="QLineEdit" name="rotate_ay">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>70</y>
         <width>161</width>
         <height>20</height>
        </rect>
       </property>
       <property name="inputMask">
        <string/>
       </property>
       <property name="text">
        <string>0.0</string>
       </property>
       <property name="placeholderText">
        <string>Вращение вокруг OY</string>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>40</y>
         <width>35</width>
         <height>10</height>
        </rect>
       </property>
       <property name="text">
        <string> °</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_22">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>70</y>
         <width>35</width>
         <height>10</height>
        </rect>
       </property>
       <property name="text">
        <string> °</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_23">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>100</y>
         <width>35</width>
         <height>10</height>
        </rect>
       </property>
       <property name="text">
        <string> °</string>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QPushButton" name="transButton">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>70</y>
      <width>141</width>
      <height>151</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>13</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Выполнить</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="name_file">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>10</y>
      <width>481</width>
      <height>31</height>
     </rect>
    </property>
    <property name="inputMask">
     <string/>
    </property>
    <property name="text">
     <string>input.txt</string>
    </property>
    <property name="placeholderText">
     <string>имя файла</string>
    </property>
   </widget>
   <widget class="QPushButton" name="readButton">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>5</y>
      <width>141</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Считать файл</string>
    </property>
   </widget>
   <widget class="QGraphicsView" name="graphicsView">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>240</y>
      <width>600</width>
      <height>550</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>600</width>
      <height>550</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>600</width>
      <height>550</height>
     </size>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>220</y>
      <width>701</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>719</width>
     <height>17</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionTransform">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Transform</string>
   </property>
   <property name="shortcut">
    <string>Return</string>
   </property>
  </action>
  <action name="actionScanFile">
   <property name="text">
    <string>ScanFile</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>move_x</tabstop>
  <tabstop>move_y</tabstop>
  <tabstop>move_z</tabstop>
  <tabstop>transButton</tabstop>
  <tabstop>tabWidget</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>transButton</sender>
   <signal>clicked()</signal>
   <receiver>actionTransform</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>669</x>
     <y>220</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>readButton</sender>
   <signal>clicked()</signal>
   <receiver>actionScanFile</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>638</x>
     <y>58</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>signal1()</signal>
  <signal>signal2()</signal>
  <slot>slot1()</slot>
  <slot>switch_mode()</slot>
  <slot>slot2()</slot>
 </slots>
</ui>
